libcaf_core/CMakeLists.txt

```diff
@@ -131,33 +127,64 @@ set(LIBCAF_CORE_SRCS
   src/work_stealing.cpp
 )
 
+# configure build_config.hpp header
+macro(to_int_value name)
+  if(${name})
+    set(${name}_INT 1)
+  else()
+    set(${name}_INT -1)
+  endif()
+endmacro()
+
+to_int_value(CAF_NO_EXCEPTIONS)
```

Comment:
Actually, this entire `to_int_value` stuff is obsolete since 3be2020441de126661cba825e485aa3c4e1c1258. Rather than moving it from one CMake file to the other, we can just drop it.